hooks/useAnalytics.ts

```diff
@@ -0,0 +1,36 @@
+import { usePostHog } from "posthog-js/react";
+
+const ANALYTICS_KEY = "local:analytics";
+
+type Analytics = { hasFirstTransaction: boolean };
+
+export default function useAnalytics() {
+  const postHog = usePostHog();
+
+  return {
+    emitOnboardingComplete: () => postHog.capture("onboarding_complete"),
+    emitFirstTransaction: async (properties: {
+      direction: "send" | "receive";
+      amount: string;
+      coin: "KAS";
+    }) => {
+      const analytics = await storage.getItem<Analytics>(ANALYTICS_KEY, {
+        fallback: {
+          hasFirstTransaction: false,
+        },
+      });
```

Comment:
It checks storage every time when sending transaction although the browser db is fast. Would it be better to cache the data by `useState`? - Not important

Reply:
I usually prefer not to optimize too early. But I agree with you in this case is so trivial better do it.

669522023b4e9efc3275f66536a76b44f29ae6a5